group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2904,6 +2908,146 @@ private void validateGenericGroupHeartbeat(
+            if (UNKNOWN_MEMBER_ID.equals(member.memberId())) {
+                if (member.groupInstanceId() != null && group.hasStaticMember(member.groupInstanceId())) {
+                    removeCurrentMemberFromGenericGroup(
+                        group,
+                        group.staticMemberId(member.groupInstanceId()),
+                        reason
+                    );
+                    memberResponses.add(
+                        new MemberResponse()
+                            .setMemberId(member.memberId())
+                            .setGroupInstanceId(member.groupInstanceId())
+                    );
+                } else {
+                    memberResponses.add(
+                        new MemberResponse()
+                            .setMemberId(member.memberId())
+                            .setGroupInstanceId(member.groupInstanceId())
+                            .setErrorCode(Errors.UNKNOWN_MEMBER_ID.code())
+                    );
+                }
+            } else if (group.isPendingMember(member.memberId())) {
+                group.remove(member.memberId());
+                timer.cancel(genericGroupHeartbeatKey(group.groupId(), member.memberId()));
+                log.info("Pending member {} has left group {} through explicit `LeaveGroup` request. Reason: {}",
+                    member.memberId(), group.groupId(), reason);
+
+                memberResponses.add(
+                    new MemberResponse()
+                        .setMemberId(member.memberId())
+                        .setGroupInstanceId(member.groupInstanceId())
+                );
+            } else {
+                try {
+                    group.validateMember(member.memberId(), member.groupInstanceId(), "leave-group");
+                    removeCurrentMemberFromGenericGroup(
+                        group,
+                        member.memberId(),
+                        reason
+                    );
+                    memberResponses.add(
+                        new MemberResponse()
+                            .setMemberId(member.memberId())
+                            .setGroupInstanceId(member.groupInstanceId())
+                    );
+                } catch (KafkaException e) {
+                    memberResponses.add(
+                        new MemberResponse()
+                            .setMemberId(member.memberId())
+                            .setGroupInstanceId(member.groupInstanceId())
+                            .setErrorCode(Errors.forException(e).code())
+                    );
+                }
+            }
+        }
+
+        CoordinatorResult<Void, Record> coordinatorResult = EMPTY_RESULT;
+        long validMemberLeaveGroups = memberResponses.stream().filter(response -> response.errorCode() == 0).count();
+        String reason = "explicit `LeaveGroup` request for " + request.members().size() + "members, for which " +
+            validMemberLeaveGroups + " were successful.";
```

Comment:
The number are not very useful here in my opinion. How about collecting the member ids to include them here? We could say something like `explicit `LeaveGroup` request for " + String.join(", ", members) + " members`